### DISCUSSION TITLE
Submitting a `NaiveDate`

### DISCUSSION THREAD

**Discussion by twiclo** [graphql-rust/juniper#1110] [Q&A] [answered]
What should be provided in the `startDate` field when creating a `NewEmployee`? Is there a sort of "now()" function?

```
input NewEmployee {
  title: String
  startDate: NaiveDate
}
```

**Comment by ilslv** [comment, marked as answer]
@twiclo no, there is no some sort of `now()` function in `GraphQL`, as it's only a communication protocol, think of it as a fancy REST. All data preparation should be done on a client side. 

Moreover `juniper` integrations don't provide a scalar named `NativeDate`, so I don't really know what format it is.

**Reply by tyranron** [reply]
@ilslv [`NaiveDate`](https://docs.rs/juniper/0.15.10/juniper/integrations/chrono/index.html) is a [`chrono::NaiveDate`] in `juniper` 0.15.

In `juniper` 0.16 (current `master`), however, it's redefined as [`Date` GraphQL scalar](https://graphql-scalars.dev/docs/scalars/date), and backed by either [`chrono::NaiveDate`] or [`time::Date`](https://docs.rs/time/0.3.15/time/struct.Date.html) on your choice.

@twiclo if you return the `NaiveDate` GraphQL scalar, you may use the whole [`chrono::NaiveDate`] API on your choice. However, if you receive it as an input, you only may make the input argument to have the default value you desired, otherwise parse the input as specified.

[`chrono::NaiveDate`]: https://docs.rs/chrono/0.4.22/chrono/naive/struct.NaiveDate.html

**Reply by ilslv** [reply]
> In juniper 0.16 (current master), however, it's redefined as [Date GraphQL scalar](https://graphql-scalars.dev/docs/scalars/date), and backed by either [chrono::NaiveDate](https://docs.rs/chrono/0.4.22/chrono/naive/struct.NaiveDate.html) or [time::Date](https://docs.rs/time/0.3.15/time/struct.Date.html) on your choice.

Forgot that we haven't released those changes yet.

**Reply by tyranron** [reply]
@ilslv I do think it's time to do so, leaving more fundamental breaking changes to 0.17.

**Reply by twiclo** [reply]
@tyranron It sounds like you're telling me for input I should take a `String` and convert that to a `NaiveDate` on the backend

**Reply by tyranron** [reply]
@twiclo yes, of course. GraphQL operates with its [built-in scalars](https://spec.graphql.org/October2021/#sec-Scalars.Built-in-Scalars) on the transport layer (especially if the transport is JSON). Any custom scalar/types exists only as a convention/semantics layer on top of built-in scalars. 

> take a `String` and convert that to a `NaiveDate` on the backend

You don't have to do it manually. Just use the `chrono::NaiveDate` Rust type on the backend, and the `juniper` will do the conversion automatically. The input on the client side, though, should be specified as a string in `YYYY-MM-DD` format (see the links above in previous comments), like `"1991-03-31"`.
